tools/libfrencutils/read_mosaic.c

```diff
@@ -80,6 +80,11 @@ int field_exist(const char* file, const char *name)
   else
     existed = 0;
 
+  status = nc_sync(ncid);
+  if(status != NC_NOERR) {
+    sprintf(msg, "field_exist: in syncing file %s.", file);
+    handle_netcdf_error(msg, status);
+  }
```

Comment:
The file associated with `ncid` is opened with `NC_NOWRITE` and doesn't need to sync before closing.  I think you can remove this logic and all other instances in this file.